src/main/kotlin/com/ajax/cryptocurrency/parser/PriceSaver.kt

```diff
@@ -0,0 +1,55 @@
+package com.ajax.cryptocurrency.parser
+
+import com.ajax.cryptocurrency.annotation.SingleShotBackgroundJob
+import com.ajax.cryptocurrency.model.Cryptocurrency
+import com.ajax.cryptocurrency.repository.CryptocurrencyRepository
+import org.json.JSONObject
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.URL
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import java.util.concurrent.ExecutorService
+
+@SingleShotBackgroundJob(startDelay = 1000, maxParallelThreads = 3)
+class PriceSaver(
+    private val cryptocurrencyRepository: CryptocurrencyRepository,
+    @Value("sleepTime") private val sleepTime: String
+) : Runnable {
+    lateinit var cryptocurrencyName: String
+    lateinit var executor: ExecutorService
+    override fun run() {
+        val url = "https://cex.io/api/last_price/$cryptocurrencyName/USD"
+
+        runCatching {
+            val btcUrl = URL(url)
+            val reader = BufferedReader(InputStreamReader(btcUrl.openStream()))
+            reader.use { bufferedReader ->
+                val line: String = bufferedReader.readLine()
+                val jsonObject = JSONObject(line)
+                val price: Float = jsonObject.getFloat("lprice")
+
+                val createdTimeStamp = OffsetDateTime.now(ZoneOffset.UTC)
+                val cryptocurrency = Cryptocurrency(
+                    null,
+                    cryptocurrencyName = cryptocurrencyName,
+                    price = price,
+                    createdTime = createdTimeStamp.toLocalDateTime()
+                )
+
+                cryptocurrencyRepository.save(cryptocurrency)
+            }
+        }.onFailure { e ->
+            logger.error("An exception occurred while saving prices", e)
+        }
+        Thread.sleep(sleepTime.toLong())
+        executor.submit(this)
```

Comment:
Seems now I understand what you doing.

For each crypto currency you fetch actual price every 5 seconds and save to DB. Correct?

Reply:
If yes then this goal can be achieved much more easier.

Java has a nice feature for that:
```java
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

public class TaskScheduler {
    public static void main(String[] args) {
        ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);

        Runnable task = () -> {
            // Your task logic here
            System.out.println("Task executed!");
        };

        scheduler.scheduleAtFixedRate(task, 0, 5, TimeUnit.SECONDS);
    }
}
```
I've asked ChatGPT to generate the code, so it could be wrong in some small details, but the idea itself definitely correct